src/components/JaegerIntegration/JaegerScatter.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { ChartWithLegend } from '@kiali/k-charted-pf4';
+import { ChartScatter } from '@patternfly/react-charts';
+import { JaegerTrace } from '../../types/JaegerInfo';
+import { isErrorTag } from './RouteHelper';
+import { PfColors } from '../Pf/PfColors';
+
+interface JaegerScatterProps {
+  traces: JaegerTrace[];
+  onClick: (idTrace) => void;
+  errorTraces?: boolean;
+}
+
+const ONE_MILLISECOND = 1000000;
+
+export class JaegerScatter extends React.Component<JaegerScatterProps> {
+  getColor(trace: JaegerTrace) {
+    const numErredSpans = trace.spans.filter(sp => sp.tags.some(isErrorTag)).length;
+    if (numErredSpans > 0) {
+      return PfColors.Red200;
+    }
+    return PfColors.Blue200;
+  }
+
+  render() {
+    const datapoints: any[] = [];
+    let seed = 1575987817125390;
+    for (let _i = 0; _i < 50; _i++) {
+      datapoints.push({
+        x: seed,
+        y: Math.random(),
+        name: `Trace ${_i}`,
+        unit: 'seconds',
+        size: 8
+      });
+      seed += 1000000;
+    }
+
+    let tracesRaw: any[] = [];
+    this.props.traces.forEach((trace, i) => {
+      tracesRaw.push({
+        x: trace.startTime,
+        y: trace.duration / ONE_MILLISECOND,
+        name: `${i}`,
+        unit: 'seconds',
+        id: trace.traceID,
+        color: this.getColor(trace),
+        size: trace.spans.length
+      } as any);
+    });
+    const data = {
+      datapoints: datapoints,
+      color: (({ datum }) => datum.color) as any,
+      legendItem: {
+        name: 'span duration'
+      } as any
+    };
+    return (
+      <ChartWithLegend
+        data={[data]}
+        unit="seconds"
+        fill={true}
+        seriesComponent={<ChartScatter />}
+        onClick={dp => alert(dp.name)}
+      />
+    );
```

Comment:
but in the Scatter chart the onClick method show the information of other point :(

Reply:
As discussed, this needs more investigation but if we don't find a fix, there's an alternative solution to compute ourselves the closest point to cursor among all the data points in chart.